src/utils/wallet.js

```diff
@@ -208,11 +208,14 @@ export class Wallet {
         this.checkNewAccount(accountId)
 
         const keyPair = nearlib.KeyPair.fromRandom('ed25519')
-        await sendJson('POST', CONTRACT_CREATE_ACCOUNT_URL, {
-            newAccountId: accountId,
-            newAccountPublicKey: keyPair.publicKey.toString()
-        })
-        await this.saveAndSelectAccount(accountId, keyPair);
+        try {
+            await sendJson('POST', CONTRACT_CREATE_ACCOUNT_URL, {
+                newAccountId: accountId,
+                newAccountPublicKey: keyPair.publicKey.toString()
+            })
+        } finally {
+            await this.saveAndSelectAccount(accountId, keyPair);
```

Comment:
this should only be done only specifically for timeout errors (either network request timeout or neacore side timeout).

all other errors shouldn't try to use account as it's clearly not created